arboretum/ibm_cloud/fetchers/fetch_cluster_resource.py

```diff
@@ -0,0 +1,154 @@
+                    'auth-provider']['config']['id-token']
+            if p.name.endswith('.pem'):
+                cluster_config.extract(name, path=self.tempdir.name)
+                ca_cert_filepath = os.path.join(self.tempdir.name, name)
+        return cluster_token, ca_cert_filepath
+
+    def _get_roks_credentials(self, cluster, api_key):
+        """Get credentials for a ROKS cluster.
+
+        This function implements the procedure described in
+        https://cloud.ibm.com/docs/openshift?topic=openshift-access_cluster#access_automation
+        """
+        s = self.session(cluster['serverURL'])
+        oauth_path = '/.well-known/oauth-authorization-server'
+        resp = s.get(oauth_path)
+        resp.raise_for_status()
+        token_endpoint = resp.json()['token_endpoint']
+        oauth_server = token_endpoint.split('/')[2]
+        s = self.session(f'https://{oauth_server}')
+        token_path = (
+            '/oauth/authorize?client_id='
+            'openshift-challenging-client&response_type=token'
+        )
+        resp = s.get(
+            token_path,
+            auth=('apikey', api_key),
+            headers={'X-CSRF-Token': 'a'},
+            allow_redirects=False
+        )
+        location = resp.headers['Location']
+        keyword = 'access_token='
+        start = location.find(keyword)
+        end = location.find('&', start)
+        cluster_token = location[start + len(keyword):end]
+        s.headers.update = ({'X-CSRF-Token': None})
+
+        return cluster_token, None
+
+    @store_raw_evidence('ibm_cloud/cluster_resources.json')
+    def fetch_cluster_resource(self):
+        """Fetch cluster resources."""
+        cluster_list_evidence = get_evidence_dependency(
+            'raw/ibm_cloud/cluster_list.json', self.locker
+        )
+        cluster_list = cluster_list_evidence.content_as_json
+        resources = {}
+        for account in cluster_list:
+            api_key = self.config.creds.get('ibm_cloud', f'{account}_api_key')
+            access_token, refresh_token = get_tokens(api_key)
+            headers = {
+                'Accept': 'application/json',
+                'Authorization': f'Bearer {access_token}',
+                'X-Auth-Refresh-Token': refresh_token
+            }
+            self.session('https://containers.cloud.ibm.com', **headers)
+
+            resources[account] = []
+            for cluster in cluster_list[account]:
+                self.session('https://containers.cloud.ibm.com', **headers)
+                config_url = f'/global/v1/clusters/{cluster["id"]}/config'
+                resp = self.session().get(config_url)
+                resp.raise_for_status()
+                cluster_config = zipfile.ZipFile(io.BytesIO(resp.content))
+                if cluster['type'] == 'kubernetes':
+                    cluster_token, ca_cert = self._get_iks_credentials(
+                        cluster_config
+                        )
```

Comment:
Please fix this.

```suggestion
                    cluster_token, ca_cert = self._get_iks_credentials(
                        cluster_config
                    )
```

Reply:
@alfinkel currently neither `make code-lint` nor `make code-format` checks this formatting.
I tried to change the behaviours and I found that changing `.pre-commit-config.yaml` as below changes the behaviour of `make code-lint` to make warning for this.

```
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -12,7 +12,9 @@ repos:
     -   id: flake8
         args: [
           --extend-ignore,
-          "P1,C812,C813,C814,C815,C816,W503"
+          "P1,C812,C813,C814,C815,C816,W503",
+          --select,
+          "E123"
         ]
         additional_dependencies: [
           flake8-2020,
```

Unfortunately I can't find how to change the behaviour of `make code-format` (yapf); `align_closing_bracket_with_visual_indent = false` in `./yapf-config` does not work as expected.

However I think it is better to use these automatic checks than manual formatting.  Could you help me find way to change the behaviour of yapf?

Reply:
We don't want to change the linter's or the formatter's behavior.  I just want you to fix your indentation.

```suggestion
                    cluster_token, ca_cert = self._get_iks_credentials(
                        cluster_config
                    )
```

Reply:
got it, I've just fixed the format of the code.